test/extended/util/pods.go

```diff
@@ -86,6 +90,31 @@ func CreateExecPodOrFail(client kubernetes.Interface, ns, name string, tweak ...
 	})
 }
 
+// CreateExecPodWaitForStatus creates a pod and checks that it achieves the given status
+// Pod name is uniquely generated.
+func CreateExecPodWaitForStatus(client kubernetes.Interface, ns, name string, phase v1.PodPhase, tweak ...func(*v1.Pod)) (bool, error) {
```

Comment:
there are certainly functions in the upstream kube e2e utils that already do this

Reply:
done